mibidata/segmentation.py

```diff
@@ -156,3 +156,54 @@ def filter_by_size(label_image, min_size, max_size):
     new_df = pd.DataFrame(df.loc[segment_labels, :]).set_index(
         pd.Index(new_labels, name='label'))
     return np.squeeze(new_image.data), new_df
+
+
+def get_adjacency_matrix(label_image):
+    """Calculates adjacency matrix.
+
+    Args:
+        label_image: An NxM array where each pixel's nonnegative integer value
+            corresponds to the label of an image region, such as a cell or
+            other segment.
+
+    Returns:
+        adjacency_matrix: NxN array of floats (N is the number of labels)
+            Each i, j element of the adjacency_matrix corresponds to the
+            fraction of the i region boundary length that is shared with j
+            region.
+
+    """
+
+    # To find the adjacent regions we stack nearest neighbors of label_image
+    # and look for pixles with more than 1 label on the stack
+
+    # To create the stack we first need to pad label_image with zeros
+    pad_image = np.pad(label_image, 1, 'constant').astype(int)
+    label_stack = np.array([
```

Comment:
I think you should note that you are defining a boundary using 4-connectivity (which is totally fine, it will just be good to note it).

Reply:
done in the comment in line 177